### PR TITLE
Master remove exception

### PR DESCRIPTION
**Pull request by dereuromark**
This seems to make more sense with outdated roles, e.g. old ones still in the session etc.

### CODE REVIEW

File: src/Auth/AclTrait.php [hunk 1]
```diff
@@ -406,7 +405,7 @@ protected function _mapped($roles) {
 			$alias = array_keys($availableRoles, $role);
 			$alias = array_shift($alias);
 			if (!$alias || !is_string($alias)) {
-				throw new RuntimeException('Cannot find role alias for role ' . $role);
+				continue;
```
Comment (dereuromark) [inline src/Auth/AclTrait.php]: Should we log this or trigger a notice?

**Comment by saeideng** [comment]
Please see line 390
that line need to removing :)
for init programs we have not even one record

**Comment by dereuromark** [comment]
You are absolutely right. I will fix this up!

**Comment by dereuromark** [comment]
So good to go?

**Comment by saeideng** [comment]
Please see again to line 347
Should we define role id?

**Comment by dereuromark** [comment]
Hm, on the one hand a user without a role always wasn't something valid so far.
But in the other hand I could understand the use case of being logged in without any roles.

So I agree, I will also remove that constraint now.

**Comment by dereuromark** [comment]
It will now just check the key, which of course must be present to make sense.
The value can be null now :) 
Thanks again for your valuable input!

Any more issues?

File: src/Auth/AclTrait.php [hunk 2]
```diff
@@ -342,10 +341,13 @@ protected function _getAvailableRoles() {
 	protected function _getUserRoles($user) {
 		// Single-role from session
 		if (!$this->_config['multiRole']) {
-			if (isset($user[$this->_config['roleColumn']])) {
-				return $this->_mapped([$user[$this->_config['roleColumn']]]);
+			if (!array_key_exists($this->_config['roleColumn'], $user)) {
+				throw new Exception(sprintf('Missing TinyAuth role id field (%s) in user session', 'Auth.User.' . $this->_config['roleColumn']));
 			}
-			throw new Exception(sprintf('Missing TinyAuth role id field (%s) in user session', 'Auth.User.' . $this->_config['roleColumn']));
+			if (!isset($user[$this->_config['roleColumn']])) {
```
Comment (saeideng) [inline src/Auth/AclTrait.php]: Duplicated with above if(!array.....  ?

Comment (dereuromark) [inline src/Auth/AclTrait.php]: No, array_key_exists is for missing key (developer error). isset() is for checking on not null values before going into the method (can be OK as you said).

Comment (saeideng) [inline src/Auth/AclTrait.php]: Looking good to me now 
thnks

**Pull request merged**